src/Raven.Client/Documents/LinqExtensions.cs

```diff
@@ -443,6 +443,45 @@ public static Task<int> CountAsync<TSource>(this IQueryable<TSource> source, Can
             return query.CountAsync(token);
         }
 
+        /// <summary>
+        /// Returns the number of elements in a sequence as int64 type.
+        /// </summary>
+        /// 
+        /// <typeparam name="TSource">
+        /// The type of the elements of source.
+        /// </typeparam>
+        /// 
+        /// <param name="source">
+        /// The <see cref="IRavenQueryable{T}"/> that contains the elements to be counted.
+        /// </param>
+        /// <param name="token">The cancellation token.</param>
+        /// 
+        /// <returns>
+        /// The number of elements in the input sequence.
+        /// </returns>
+        /// 
+        /// <exception cref="ArgumentNullException">
+        /// source is null.
+        /// </exception>
+        /// 
+        /// <exception cref="OverflowException">
+        /// The number of elements in source is larger than <see cref="Int32.MaxValue"/>.
+        /// </exception>
+        public static Task<long> CountLongAsync<TSource>(this IQueryable<TSource> source, CancellationToken token = default)
```

Comment:
I think it would be better to have LongCountAsync. Matches what .NET has: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.longcount?view=net-5.0

Reply:
done